### ISSUE TITLE
Cache builder responses

### ISSUE THREAD

**Issue by lmossman** [airbytehq/airbyte#27292]
## Context

During a build-a-connector session, we realized that data doesn't actually need to be re-fetched from the source API when only fields that process the data are changed (e.g. record selector or primary key)

In this case, the previously-fetched data can be reused, and just the post-processing can be re-performed.

Probably the most straightforward path here would be to cache responses in the server.

## Acceptance Criteria
Changes to the manifest which don't change the outgoing requests do not cause a re-fetching of data from the source

**Comment by flash1293** [comment]
I don't think we have sticky sessions, so a different pod could serve subsequent requests, right? Still might help OSS instances with just a single builder server pod
